### PR TITLE
Cause total tracker

### PR DESCRIPTION
**Pull request by jordycollingwood**
Incorporated features such as email address, thumbnail and cause total tracker into contract info struct.

Converted end cause functions to a single toggle function, utilising 1 and 2 instead of bool for gas fee optimisation.

### CODE REVIEW

**Comment by finnformica** [comment]
Could you sort out the merge conflicts before I review the code? Theres a button below this comment called 'Resolve conflicts' that provides an easy GUI for it

File: CauseContract.sol [hunk 1]
```diff
@@ -50,29 +86,55 @@ contract CauseContract {
     }
 
     function retrieveInfo() public view returns (ContractInfo memory) {
-        return ContractInfo(id, admin, incoming, outgoing, contractAddress);
+        return ContractInfo(id, admin, incoming, outgoing, contractAddress, causeTotal, endCause, emailAddress, causeDescription, websiteURL, thumbnail);
```
Comment (finnformica) [inline CauseContract.sol]: Could you change thumbnail to thumbnailURL just to make it clear what we should be expecting from it?

File: CauseContract.sol [hunk 2]
```diff
@@ -50,29 +86,55 @@ contract CauseContract {
...
     }
 
-    uint256 public transactionFee;
-    
-    function donate() public payable returns (uint256) {
+
+    function donate() public payable returns (bool) {
```
Comment (finnformica) [inline CauseContract.sol:93]: Does this need to return true?

File: CauseContract.sol [hunk 3]
```diff
@@ -50,29 +86,55 @@ contract CauseContract {
...
+
         require(msg.value > 0, "You must send some Ether");
+        require(endCause == 1, "This cause has ended, your funds have been returned");
+
+        uint256 gasStart = gasleft();
+
+        transactionFee = (msg.value*BASIS_POINTS) / 1000; // Transaction fee of 5bps (by default)
+        
+
+        blockChange.transfer(transactionFee);
```
Comment (finnformica) [inline CauseContract.sol]: Which method of fund transfer did we decide was better, .transfer() or .call() - could this be updated

Comment (jordycollingwood) [reply CauseContract.sol]: I believe call is best in this instance, but for our purposes they are equivalent, call just gives us more flexibility.

**Review by jordycollingwood — COMMENTED** [review]
Changed transfer method to call

**Review by finnformica — APPROVED** [review]

File: CauseContract.sol [hunk 4]
```diff
@@ -25,23 +41,43 @@ contract CauseContract {
...
+
     // contract information struct
     struct ContractInfo {
         string id;
         address admin;
         Transaction[] incoming;
         Transaction[] outgoing;
         address contractAddress;
+        uint256 causeTotal;
+        uint256 endCause;
```
Comment (finnformica) [inline CauseContract.sol:66]: Ah, one last thing could you change endCause to causeState just so its clearer when integrating the front-end, ty

**Pull request merged**